### PR TITLE
#167190433 Implement input validations

### PR DESCRIPTION
**Pull request by chuxmykel**
#### What does this PR do?
- This PR integrates the `@hapi/joi` validation library with the app to validate input.

#### Description of the task to be completed 
- install `@hapi/joi` library for input validation
- validate sign in and sign up routes

#### How should this be manually tested?
- visit the staging link for this PR and try signing up with invalid data.

#### What are the relevant Pivotal Tracker stories?
- [#167190433](https://www.pivotaltracker.com/story/show/167190433)

### CODE REVIEW

**Review by encodedBicoding — APPROVED** [review]
LGTM

**Review by cvjude — APPROVED** [review]
LGTM

File: server/routes/user.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import express from 'express';
 import UserController from '../controllers/userController';
-
+import InputValidator from '../middlewares/inputValidator';
 
 const userRoute = express();
```
Comment (Lundii) [inline server/routes/user.js:6]: I think you can use destructuring here. Something like 
```
const { validateUser, validateLogin } = InputValidator;
const { signUp, login } = UserController;
```
and then 
`userRoute.post('/', validateUser, signup)`.

What do you think.

Comment (chuxmykel) [reply server/routes/user.js:6]: Nice. Good idea. I will do that now. Thanks.

Comment (Lundii) [reply server/routes/user.js:6]: > Nice. Good idea. I will do that now. Thanks.

Good job

**Review by Lundii — APPROVED** [review]

**Review by chiboycalix — APPROVED** [review]
Looks good to me. Awesome job @chuxmykel

**Review by Hadeneekeh — APPROVED** [review]

**Comment by Hadeneekeh** [comment]
LGTM

**Review by obayomi96 — APPROVED** [review]
Good one @chuxmykel , you did a fine job

**Pull request merged**